### ISSUE TITLE
Bug: unable to locate index metadata for existent sxd-xpath crate

### ISSUE THREAD

**Issue by SVilgelm** [EmbarkStudios/cargo-deny#630]
### Describe the bug

```
warning[index-failure]: unable to check for yanked crates
    ┌─ ...\Cargo.lock:151:1
    │
151 │ sxd-xpath 0.4.2 registry+https://github.com/rust-lang/crates.io-index
    │ --------------------------------------------------------------------- crate whose registry we failed to query
    │
    = unable to locate index metadata
    = sxd-xpath v0.4.2
```

### To reproduce

* add `sxd-xpath` as dependency 
* run `cargo deny check all`

### cargo-deny version

cargo-deny 0.14.16

### What OS were you running cargo-deny on?

Windows

### Additional context

the 0.4.2 version is present in the db: https://github.com/rust-lang/crates.io-index/blob/master/sx/d-/sxd-xpath#L9

but the crate is not present in the advisory-dbs folder: CARGO_HOME\advisory-dbs\github.com-2f857891b7f43c59\crates

![image](https://github.com/EmbarkStudios/cargo-deny/assets/523825/ea3f078a-b446-4a45-93e7-abcf2e0e4dd3)

**Comment by SVilgelm** [comment]
@Jake-Shadle Do you know how can I debug this issue, because it is constant and reproducible every time on my laptop?

**Comment by SVilgelm** [comment]
Okay, i've figured out. I work for Microsoft, and due to security rules, we cannot fetch the crates from crates.io, instead we have to use the Azure DevOpst feeds, like an internal mirror. So, the `.cargo/config.toml` file has an override for `crates-io` source:
```toml
[registries]
WlmOxide = { index = "sparse+https://.../WlmOxide/Cargo/index/" }

[source.crates-io]
replace-with = "WlmOxide"
```

which means, that `cargo` tool does not fetch anything from `crates.io`, but fetches from the internal registry. And the `deny` worked only because I cloned some other projects on my laptop and build them, so `cargo` was able to fetch some crates from `crates.io`. And the issue I had was because of the outdated index.
No `deny` work fine, I commented out those line, run `cargo update --dry-run` to fetch the crates from `crates.io` :)
